src/textual/_config.py

```diff
@@ -0,0 +1,207 @@
+        should be removed."""
+        try:
+            with open(self._default_config_path, "rb") as default_config_file:
+                default_config = tomli.load(default_config_file)
+        except OSError:
+            return {}
+
+        default_config = _filter_keys(
+            default_config, keys_to_keep=self._default_config_headers
+        )
+        return default_config
+
+    def _read_and_filter_user_config_files(self) -> list[dict[str, object]]:
+        """Read the config files on the users machine. Keys found in here inside matching namespaces will ultimately
+        override corresponding keys inside the default config files."""
+        user_configs = []
+        for path in self._user_config_paths:
+            try:
+                with open(path, "rb") as user_config_file:
+                    user_config = tomli.load(user_config_file)
+            except OSError:
+                continue
+
+            user_config = _filter_keys(
+                user_config, keys_to_keep=self._user_config_headers
+            )
+            user_configs.append(user_config)
+
+        return user_configs
+
+    @property
+    def _default_config_headers(self) -> frozenset[str]:
+        """Get a frozenset of strings containing the names of the section headers relevant to the default config that
+        is bundled with this application (*not* for user config)."""
+        namespace = self.namespace
+        return frozenset((f"{namespace}",))
+
+    @property
+    def _user_config_headers(self) -> frozenset[str]:
+        """Get a frozenset of strings containing the names of the section headers relevant to the user config that
+        is bundled with this application."""
+        return frozenset(
+            (
+                "meta",
+                "textual",
+                *self._default_config_headers,
+            )
+        )
+
+    def _merge_raw_config_dicts(
+        self,
+        default_configs: dict[str, object],
+        user_configs: Iterable[dict[str, object]],
+    ) -> dict[str, object]:
+        """Merge user config dictionaries into the default config dictionary"""
+        merged = default_configs.copy()
+
+        for user_config in user_configs:
+            # Update the individual sections that should be merged
+            merged_namespace = merged.get(self.namespace, {})
+            user_config_namespace = user_config.get(self.namespace, {})
+
+            # By "sandbox" we refer to `APP_NAME.config` section
+            merged_config_sandbox = merged_namespace.get("config", {})
+            user_config_sandbox = user_config_namespace.get("config", {})
+
+            # Merge the `meta` section
+            merged_meta = merged.get("meta", {})
+            merged_meta.update(user_config.get("meta", {}))
+
+            # Merge the `textual` section
+            merged_textual = merged.get("textual", {})
```

Comment:
My concern here is that we only merge sections we know about. 

Third party libs may not get such merging behaviour by default. Is that right?

Reply:
3rd party apps would get the same merging behaviour. They'll have a namespace and if the same config appears in that namespace in the users own config file, items will be replaced by the users values.

Reply:
But if an app section adds a sub-dict, those wouldn't be merge?

Something like this:

```
# default.toml
[app.foo]
bar = {a = 1}

# user.toml
[app.foo]
bar = {b = 1}
```

App.foo.bar would be `{b=1}` right?